connector/dialect.tdd

```diff
@@ -1,376 +1,147 @@
+      <argument type='int' />
     </function>
-    <function group='string' name='UPPER' return-type='str'>
-      <formula>UPPER(CAST(%1 AS TEXT))</formula>
+    <function group='aggregate' name='MIN' return-type='str'>
+      <formula>MIN(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
       <argument type='str' />
     </function>
-    <function group='date' name='DAY' return-type='int'>
-      <formula>CAST(EXTRACT(DAY FROM %1) AS INTEGER)</formula>
-      <argument type='datetime' />
-    </function>
-    <function group='date' name='MAX' return-type='datetime'>
-      <formula>(CASE
-	WHEN %1 IS NULL OR %2 IS NULL THEN NULL
-	WHEN %1 &gt; %2 THEN %1
-	ELSE %2 END)</formula>
-      <argument type='datetime' />
-      <argument type='datetime' />
-    </function>
-    <function group='date' name='MAX' return-type='date'>
-      <formula>(CASE
-	WHEN %1 IS NULL OR %2 IS NULL THEN NULL
-	WHEN %1 &gt; %2 THEN %1
-	ELSE %2 END)</formula>
-      <argument type='date' />
-      <argument type='date' />
-    </function>
-    <function group='date' name='MIN' return-type='datetime'>
-      <formula><![CDATA[(CASE
-	WHEN %1 IS NULL OR %2 IS NULL THEN NULL
-	WHEN %1 < %2 THEN %1
-	ELSE %2 END)]]></formula>
-      <argument type='datetime' />
-      <argument type='datetime' />
-    </function>
-    <function group='date' name='MIN' return-type='date'>
-      <formula><![CDATA[(CASE
-	WHEN %1 IS NULL OR %2 IS NULL THEN NULL
-	WHEN %1 < %2 THEN %1
-	ELSE %2 END)]]></formula>
-      <argument type='date' />
-      <argument type='date' />
-    </function>
-    <function group='date' name='MONTH' return-type='int'>
-      <formula>CAST(EXTRACT(MONTH FROM %1) AS INTEGER)</formula>
-      <argument type='datetime' />
-    </function>
-    <function group='date' name='NOW' return-type='datetime'>
-      <formula>CURRENT_TIMESTAMP</formula>
-    </function>
-    <function group='date' name='QUARTER' return-type='int'>
-      <formula>CAST(EXTRACT(QUARTER FROM %1) AS INTEGER)</formula>
+    <function group='aggregate' name='MIN' return-type='datetime'>
+      <formula>MIN(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
       <argument type='datetime' />
     </function>
-    <function group='date' name='TODAY' return-type='date'>
-      <formula>CURRENT_DATE</formula>
-    </function>
-    <function group='date' name='WEEK' return-type='int'>
-      <formula>CAST(FLOOR((7 + EXTRACT(DOY FROM %1) - 1 + EXTRACT(DOW FROM DATE_TRUNC(&apos;YEAR&apos;, %1))) / 7) AS INTEGER)</formula>
-      <argument type='datetime' />
+    <function group='aggregate' name='SUM' return-type='real'>
+      <formula>SUM(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
+      <argument type='real' />
     </function>
-    <function group='date' name='YEAR' return-type='int'>
-      <formula>CAST(EXTRACT(YEAR FROM %1) AS INTEGER)</formula>
-      <argument type='datetime' />
+    <function group='aggregate' name='SUM' return-type='int'>
```

Comment:
We support `SUM(DISTINCT %1)`. Should we add that to this file?

Reply:
I added it as `SUMD`